pkg/importer/dataStream_test.go

```diff
@@ -25,6 +46,78 @@ func createDataStream(ep, accKey, secKey string) *dataStream {
 	return ds
 }
 
+func createDataStreamMulti(testfile, ep string) *dataStream {
+	// set our directory for images
+	imageDir, _ := filepath.Abs("../../test/images")
+	urlPath := filepath.Join(ep, testfile)
+	testFilePath := filepath.Join(imageDir, testfile)
+
+	dsurl, _ := ParseEndpoint(urlPath)
+
+	ds := &dataStream{
+		Url:         dsurl,
+		buf:         make([]byte, image.MaxExpectedHdrSize),
+		accessKeyId: "",
+		secretKey:   "",
+	}
+
+	ds.constructReaders()
+
+	buf := bytes.NewBuffer(nil)
+	f, _ := os.Open(testFilePath)
+	io.Copy(buf, f)
+	f.Close()
+	testBytes, _ := ioutil.ReadAll(buf)
+
+	//append readers
+	xzreader, _ := xz.NewReader(bytes.NewReader(testBytes), 0)
+	tarreader := tar.NewReader(xzreader)
+
+	ds.appendReader(RdrXz, xzreader)
+	ds.appendReader(RdrTar, tarreader)
+
+	return ds
+}
+
+func createDataStreamBuf(ep, accKey, secKey, testfile string) (*dataStream, []byte, int) {
```

Comment:
In quickly reviewing the code this func is pretty similar to `createDataStreamMulti`. If 2 similar funcs are needed then consider having them share common helper funcs. Most of my comments in `createDataStreamMulti` apply here (which is one reason to use helpers!)